### PR TITLE
Copter: Determine the control mode from the flight mode pointer

### PR DESCRIPTION
**Pull request by muramura**
I set the control_mode value to the flightmode pointer.
I think, do not set the flightmode pointer different from the control_mode value.

### CODE REVIEW

File: ArduCopter/Copter.cpp [hunk 1]
```diff
@@ -656,12 +656,13 @@ Copter::Copter(void)
...
     inertial_nav(ahrs),
-    param_loader(var_info),
-    flightmode(&mode_stabilize)
+    param_loader(var_info)
 {
     // init sensor error logging flags
     sensor_health.baro = true;
     sensor_health.compass = true;
+
+    flightmode = mode_from_mode_num(control_mode);
```
Comment (peterbarker) [inline ArduCopter/Copter.cpp:665]: This should be the other way around, setting control_mode from flight_mode.

**Review by peterbarker — COMMENTED** [review]
This will conflict with https://github.com/ArduPilot/ardupilot/pull/13431/files - and that PR is a much, much better way to fix this issue!

File: ArduCopter/Copter.cpp (hunk 1 above)
Comment (muramura) [reply ArduCopter/Copter.cpp:665]: @peterbarker san.
Thanks for your comment.
I will add the mode_from_mode_num method mode_num_from_mode method.
I will try to get the mode number from the mode pointer.

Comment (muramura) [reply ArduCopter/Copter.cpp]: I changed the flight mode pointer to set the control mode.

**Comment by peterbarker** [comment]
I've merged that other PR from @tatsuy which eliminates `control_mode` entirely - thanks anyway!

**Pull request closed without merging**